src/main/java/LinkedQueue.java

```diff
@@ -77,6 +78,35 @@ public String toString() {
         return builder.toString();
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        LinkedQueue<?> that = (LinkedQueue<?>) o;
+        if (this.size == that.size) {
+            Node<?> thisCurrent = this.front;
+            Node<?> thatCurrent = that.front;
+            while (thisCurrent != null) {
+                if (!thisCurrent.data.equals(thatCurrent.data)) return false;
+                thisCurrent = thisCurrent.getNext();
+                thatCurrent = thatCurrent.getNext();
+            }
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int hash = Objects.hashCode(size);
+        Node<?> current = front;
+        while (current != null) {
+            hash += Objects.hashCode(current.getData());
```

Comment:
I'm not sure if you're being lazy with hashcodes as I think I've mentioned this a few times, but this is not the ideal way to create a hashcode. Simply adding components will create a lot of hash collisions.

You should also be multiplying the existing hash by a prime number. See [String#hashCode](https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/StringUTF16.java#L414-L421) for an example. The shift causes very similar objects to have very different hashcodes.